index.js

```diff
@@ -16,6 +16,7 @@ const byline = require('byline')
 const resolveFrom = require('resolve-from')
 
 const DEFAULT_NODE_GYP_PATH = resolveFrom(__dirname, 'node-gyp/bin/node-gyp')
+const _hookExistsCache = {}
```

Comment:
I'd rather this be a `new Set()`, since that's gonna be pretty efficient for this use-cases. There's a good chance we might be hammering this cache a lot.

Reply:
has that been benchmarked? for string keys, `Object.create(null)` might actually be faster than a Set.

Reply:
I need to cache positive and negative values, so `Set` seems inappropriate here. Also, the perf overhead here is peanuts, but I wouldn’t mind switching it out for `Object.create(null)` if that satisfies the micro optimization gods :-)

Reply:
Or I can use Map, but all-in-all I think any of the 3 choices is ok. Make the call and I’ll switch it.

Reply:
A Map is conceptually correct since you need more than just presence; a null object is likely to be faster based on recent benchmarks on a recent aphrodite PR - either seems fine to me, but I’d suggest not using a normal object.